source/extensions/filters/network/redis_proxy/command_splitter_impl.cc

```diff
@@ -447,10 +452,135 @@ bool areAllResponsesSame(const std::vector<Common::Redis::RespValuePtr>& respons
+                          if (integerValue >= 0) {
+                              sum += integerValue; // Add integer value to sum if it's non-negative
+                          } else {
+                              // Handle negative integer value
+                              std::cerr << "Error: Negative integer value: " << integerValue << std::endl;
+                              ENVOY_LOG(error, "Error: Negative integer value: {}", integerValue);
+                          }
+                      } catch (const std::exception& e) {
+                          // Handle conversion error
+                          std::cerr << "Error converting integer: " << e.what() << std::endl;
+                          ENVOY_LOG(error, "Error converting integer: {}", e.what());
+                      }
+                    }
+                  }
+                  response->asInteger() = sum;
+                  callbacks_.onResponse(std::move(response));
+                  pending_responses_.clear();
+              } else {
+                  Common::Redis::RespValuePtr response = std::make_unique<Common::Redis::RespValue>();
+                  Common::Redis::RespValue innerResponse;
+                  response->type(Common::Redis::RespType::Array);
+
+                  // Iterate through pending_responses_ and append non-empty responses to the array
+                  if ( pending_requests_[reqindex].redisarg_ == "channels" || pending_requests_[reqindex].rediscommand_ == "keys" || pending_requests_[reqindex].redisarg_ == "get") {
+                      for (auto& resp : pending_responses_) {
+                        if (resp->type() == Common::Redis::RespType::Array) {
+                          if (resp->type() == Common::Redis::RespType::Array && resp->asArray().empty()) {
+                            continue; // Skip empty arrays
+                          }
+                          innerResponse = std::move(*resp); // Move the content to innerResponse
+                          if (innerResponse.type() == Common::Redis::RespType::Array) {
+                              for (auto& elem : innerResponse.asArray()) {
+                                  response->asArray().emplace_back(std::move(elem));
+                              }
+                          } else {
+                              // Append non-array responses directly
+                              response->asArray().emplace_back(std::move(innerResponse));
+                          }
+                        } 
+                      }
+                  } else if (pending_requests_[reqindex].redisarg_ == "numsub"){
+                      std::unordered_map<std::string, int> subscriberCounts;
+                      for (auto& resp : pending_responses_) {
+                        if (resp->type() == Common::Redis::RespType::Array) {
+                            innerResponse = std::move(*resp);
```

Comment:
why is move required here?
this makes the object in pending_responses_ in a unspecifed state, which is not good enough to be used later